client.js

```diff
@@ -268,5 +269,10 @@ Client.prototype._defaultAnnounceOpts = function (opts) {
   if (opts.left == null && self.torrentLength != null) {
     opts.left = self.torrentLength - opts.downloaded
   }
+
+  if (opts.getAnnounceOpts == null) {
+    opts.getAnnounceOpts = self._getAnnounceOpts
+  }
```

Comment:
This isn't necessary. All tracker types already have a reference to the `Client` instance. So they should check for `self.client._getAnnounceOpts`.